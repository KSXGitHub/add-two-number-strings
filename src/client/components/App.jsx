import React from 'react'
import AppBar from 'material-ui/AppBar'
import Paper from 'material-ui/Paper'
import DefaultMuiTheme from './DefaultMuiTheme.jsx'
import RandomNumber from './RandomNumber.jsx'

export default function App ({init}) {
  const display = num =>
    parseFloat(num).toString(16).slice(2, 18)

  return <div>
    <DefaultMuiTheme><div>
      <AppBar title='Hello, World!!' />
      <Paper zDepth={1}>
        <RandomNumber display={display} init={init} />
      </Paper>
    </div></DefaultMuiTheme>
  </div>
}
