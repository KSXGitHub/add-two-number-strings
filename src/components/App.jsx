import React from 'react'
import AppBar from 'material-ui/AppBar'
import Paper from 'material-ui/Paper'
import Toggle from 'material-ui/Toggle'
import MuiWrapper from './MuiWrapper.jsx'
import RandomNumberForm from './RandomNumberForm.jsx'
import ClockForm from './ClockForm.jsx'
import Footer from './Footer.jsx'

export const TOP_PAPER_STYLE = {
  height: '100%',
  padding: 0,
  border: 0,
  margin: 0
}

export default class App extends React.Component {
  constructor (props) {
    super(props)

    const {
      init,
      darkTheme = false
    } = props

    this.state = {init, darkTheme}
  }

  render () {
    const {init, darkTheme} = this.state

    return <MuiWrapper darkTheme={darkTheme}><Paper style={TOP_PAPER_STYLE}>
      <header>
        <AppBar title='Hello, World!!' showMenuIconButton={false} />
      </header>
      <main>
        <RandomNumberForm init={init} />
        <ClockForm darkTheme={darkTheme} />
        <Paper zDepth={0}>
          <Toggle
            label='Dark Theme'
            toggled={darkTheme}
            onToggle={(_, darkTheme) => this.setState({darkTheme})}
          />
        </Paper>
      </main>
      <Footer />
    </Paper></MuiWrapper>
  }
}
