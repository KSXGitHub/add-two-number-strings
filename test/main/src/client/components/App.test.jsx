import React from 'react'
import renderer from 'react-test-renderer'
import App from '../../../../../src/client/components/App.jsx'

test('Snapshot: components/App.jsx', () => {
  const init = () => 0.123456789
  const tree = renderer.create(<App init={init} />).toJSON()
  expect(tree).toMatchSnapshot()
})
