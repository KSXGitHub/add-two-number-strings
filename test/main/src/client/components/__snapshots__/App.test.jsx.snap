// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: components/App.jsx 1`] = `
<div>
  <div>
    <span
      className="label"
    >
      Random Number: 
    </span>
    <span
      className="value"
    >
      67ab265d7da3
    </span>
  </div>
</div>
`;
